### ISSUE TITLE
Multiple calls to CPLEX.free_problem gives a segfault

### ISSUE THREAD

**Issue by trigaut** [JuliaOpt/CPLEX.jl#234]
@MarcMush notified me that when trying to free many CPLEX problems with the function  `CPLEX.free_problem` we always run into a Segmentation Fault.
https://github.com/JuliaOpt/CPLEX.jl/blob/0ab8e859e05d07f3230d6375b929413eb1d3d291/src/cpx_model.jl#L227

The following example always crashes Julia after 200 or 300 iterations on my computer (Ubuntu 16.04, Julia 1.1, CPLEX1290, Last tagged CPLEX.jl) with a signal(7) : Bus error.

```julia
using JuMP, CPLEX

function build_direct_model()
    m = direct_model(CPLEX.Optimizer(CPX_PARAM_SCRIND=0))
    @variable(m, 0 <= x <= 1)
    @objective(m, Min, x)
    return m
end


function run_direct_models()
	k=0

	while true
		k+=1
		println(k)
		m = build_direct_model()
		CPLEX.free_problem(m.moi_backend.inner)
		sleep(0.1)
	end
end 

run_direct_models()
```
Same issue with caching JuMP models instead of direct models (after a call to `optimize!`).

The following fix seems to solve the issue and still free the memory.

```julia
function free_problem(model::Model)
    tmp = model.lp # instead of Ptr{Cvoid}[model.lp]
    stat = @cpx_ccall(freeprob, Cint, (Ptr{Cvoid}, Ptr{Cvoid}), model.env.ptr, tmp)
    if stat != 0
        throw(CplexError(model.env, stat))
    end
end
```

Or did we get something wrong?

**Comment by odow** [comment]
Your fix seems correct. The `Ptr{Cvoid}[model.lp]` is just an artefact from 5 years ago: https://github.com/JuliaOpt/CPLEX.jl/commit/4d8f78fabd36daea6e8ec6ca6317258d457925b8

**Comment by odow** [comment]
Closed by #235
